openff/qcsubmit/validators.py

```diff
@@ -288,14 +286,26 @@ def check_environments(environment: str) -> str:
     """
 
     # try and make a new chemical environment checking for parse errors
-    _ = ChemicalEnvironment(smirks=environment)
-
-    # check for numeric tags in the environment
-    if re.search(":[0-9]]+", environment) is not None:
-        return environment
-
-    else:
-        raise SMIRKSParsingError(
-            "The smarts pattern passed had no tagged atoms please tag the atoms in the "
-            "substructure you wish to include/exclude."
-        )
+    try:
+        _ = Molecule().chemical_environment_matches(environment)
+        # check for numeric tags in the environment
+        if re.search(":[0-9]]+", environment) is not None:
+            return environment
```

Comment:
(not blocking, maybe future issue) I don't understand why we have this check. From what I understand the purpose of the SMIRKS filter in QCSubmit to be - that is, returning True or False on entire molecules - A SMIRKS without numeric tags should be valid.

Reply:
Yeah I'm not sure about that one either. I just carried it over because it was there before! Did you decide we do want to keep this check for now too? I left that in the Questions section at the top too.

Reply:
I decided to keep it. It's making this check slightly more strict then it needs to be, but

* This entire check is just a performance hack/UX improvement - without this check an invalid SMIRKS would just error out later when it's actually being used
* There may be some cases that I'm unaware of where removing this would break something, and it's possible that those cases aren't tested so CI wouldn't save us.
* The possibly-excessive requirement must not be too burdensome on users because nobody is complaining about it

Reply:
> From what I understand the purpose of the SMIRKS filter in QCSubmit to be - that is, returning True or False on entire molecules

Ah, I see, turns out this assumption is wrong. This method is used in [several places](https://github.com/openforcefield/openff-qcsubmit/blob/2461c47b63be2a5c328caa688bb59aa07b402be3/openff/qcsubmit/workflow_components/fragmentation.py#L25-L39) where there must be tagged atoms. So removing this would have been a regression. Good thing we didn't do that :-)